code/modules/mob/living/carbon/martian/inventory.dm

```diff
@@ -0,0 +1,157 @@
+
+/mob/living/carbon/martian/get_item_offset_by_index(index)
+	switch(index)
+		if(1,6)
+			return list("x"=0, "y"=0)
+		if(2,5)
+			return list("x"=0, "y"=10)
+		if(3,4)
+			return list("x"=0, "y"=16)
+
+	return list()
+
+/mob/living/carbon/martian/get_held_item_ui_location(index)
+	if(!is_valid_hand_index(index))
+		return
+
+	switch(index)
+		if(1)
+			return "CENTER-3:16,SOUTH:5"
+		if(2)
+			return "CENTER-2:16,SOUTH:5:4"
+		if(3)
+			return "CENTER-1:16,SOUTH:5:10"
+		if(4)
+			return "CENTER+1:16,SOUTH:5:10"
+		if(5)
+			return "CENTER+2:16,SOUTH:5:4"
+		if(6)
+			return "CENTER+3:16,SOUTH:5"
+		else
+			return ..()
+
+/mob/living/carbon/martian/get_index_limb_name(index)
+	if(!index)
+		index = active_hand
+
+	switch(index)
+		if(1)
+			return "right lower tentacle"
+		if(2)
+			return "right middle tentacle"
+		if(3)
+			return "right upper tentacle"
+		if(4)
+			return "left upper tentacle"
+		if(5)
+			return "left middle tentacle"
+		if(6)
+			return "left lower tentacle"
+		else
+			return "tentacle"
+
+/mob/living/carbon/martian/get_direction_by_index(index)
+	if(index <= 3)
+		return "right_hand"
+	else
+		return "left_hand"
+
+
+/mob/living/carbon/martian/GetAccess()
+	var/list/ACL=list()
+
+	for(var/obj/item/I in held_items)
+		ACL |= I.GetAccess()
+
+	return ACL
+
+/mob/living/carbon/martian/get_visible_id()
```

Comment:
Do doors and vending machines not use this then?

Reply:
That one probably works then.